ParcialProgramacion4/src/Puesto.java

```diff
@@ -6,11 +6,39 @@ public abstract class Puesto {
     private ArrayList<Convocatoria>convocatorias;
     private ArrayList<Empleado>empleados;
 
+    public abstract boolean esJerarquico();
```

Comment:
No hace falta que sea abstracto.

Se puede definir con un  [this.getClass().getName().equals("PuestoJerarquico")]

De esta forma la llamada va siempre al padre y devuelve directamente true o false sin tener que ir a los hijos